### PR TITLE
fix: add bounds checking to the selectedrow func

### PR DESCRIPTION
**Pull request by MikaelFangel**
### Changes:
 * This adds bounds checking to the function returning the row under the cursor. By adding the check, it avoids that the program crashes when calling the function and the cursor is out of bounds.

### Still an issue:
 * The cursor can still become out of bounds values, which is a problem in itself.

### CODE REVIEW

**Comment by muesli** [comment]
Looks reasonable. Also noticed that `m.cursor` can become negative during debugging.

**Comment by muesli** [comment]
@MikaelFangel Note the linter warning.

**Comment by MikaelFangel** [comment]
@muesli yep, just noticed and have fixes it now.

File: table/table.go [hunk 1]
```diff
@@ -266,7 +266,11 @@ func (m *Model) UpdateViewport() {
 // SelectedRow returns the selected row.
 // You can cast it to your own implementation.
 func (m Model) SelectedRow() Row {
-	return m.rows[m.cursor]
+	// Check if the cursor is within bounds else return  nil
+	if m.cursor < len(m.rows) && m.cursor >= 0 {
```
Comment (muesli) [inline table/table.go:270]: May I suggest swapping the logic around? We should check for `m.cursor` being out of bounds as the exception and keep the "happy path" on the left edge.

Comment (MikaelFangel) [reply table/table.go:270]: You mean? 
```suggestion
	if m.cursor >= 0 && m.cursor < len(m.rows) {
```

File: table/table.go [hunk 2]
```diff
@@ -266,7 +266,11 @@ func (m *Model) UpdateViewport() {
...
+		return m.rows[m.cursor]
+	}
+	return nil
```
Comment (MikaelFangel) [inline table/table.go:273]: Like this?
```suggestion
	if !(m.cursor >= 0 && m.cursor < len(m.rows)) {
		return nil
	}
	return m.rows[m.cursor]
```

File: table/table.go (hunk 1 above)
Comment (MikaelFangel) [reply table/table.go:270]: Great suggestion.

File: table/table.go (hunk 2 above)
Comment (muesli) [reply table/table.go]: I'd probably go with:

```go
if m.cursor < 0 || m.cursor >= len(m.rows) {
    return nil
}

return m.rows[m.cursor]
```

Should be a little more digestible for future readers.

Comment (MikaelFangel) [reply table/table.go]: I see I've added a commit with the suggested change. Thank you for the comments. (:

Comment (MikaelFangel) [reply table/table.go]: And sorry for the mess with all the commits 😅

**Comment by muesli** [comment]
Thank you, @MikaelFangel!

**Pull request merged**